File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.java
class Solution {
    public int maxDistance(int[] colors) {
        int len = colors.length;
        int pL = 0;
        int pR = len - 1;
        int first = colors[0];
        int last = colors[len - 1];
        while (pR > 0) {
            if (colors[pR] != first) break;
            else pR--;
        }
        while (pL < colors.length) {
            if (colors[pL] != last) break;
            else pL++;
        }
        return Math.max(len - pL - 1, pR);
    }
}

File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.kt
class Solution {
    fun maxDistance(colors: IntArray): Int {
    var left = 0
    var right = colors.size-1
    var dist = 0
    var max = 0
	
    while(left<right) {
        if (colors[left]!=colors[right]) {
            if (dist<right-left && right-left>max) {
                max = right-left
            } else if (dist>=right-left && dist>max){
                max = dist
            }
        }
        right--
        dist++
    }
    return max
}
}